FAIR.md

```diff
@@ -0,0 +1,32 @@
+# Why this Fork
+In a mutual TLS-disabled multicluster Kubernetes setup, gRPC liveness and readiness probes don't work in remote clusters. However, this behavior could not be replicated on the Kubernetes cluster running the Istio control plane.
+
+We verified that the remote proxy sidecar containers had connectivity to Pilot Discovery. However, we noticed when comparing the logs in the sidecar containers on a remote cluster vs the local cluster that the local cluster sidecars were properly registering liveness and readiness probes, and the remote clusters weren't.
+
+```bash
+# GOOD - On the "local" control plane - we expect to see a listener registered for 8080
+[2019-03-06 21:44:39.033][15][info][upstream] external/envoy/source/server/lds_api.cc:80] lds: add/update listener '10.21.3.242_8080'<Paste>
+
+# BAD - On the "remote" cluster - where are 3333 and 9999 coming from???
+[2019-03-06 22:00:16.701][10][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:494] add/update cluster inbound|3333||mgmtCluster during init
+[2019-03-06 22:00:16.702][10][info][upstream] external/envoy/source/common/upstream/cluster_manager_impl.cc:494] add/update cluster inbound|9999||mgmtCluster during init
+```
+
+After diving into the Istio source, we only found that 3333 and 9999 were ports returned from the [Envoy debug registry](https://github.com/istio/istio/blob/1.0.2/pilot/pkg/proxy/envoy/v2/debug.go#L324-L337).
+
+In the Istio logs, we could see that a remote cluster registry was being picked up. However, based off of the [server bootup order]*(https://github.com/istio/istio/blob/1.0.2/pilot/pkg/bootstrap/server.go#L211-L219) (note that Discovery service is always init'd before Multicluster), it looked like the debug registry would always be loaded before the multicluster ones. Since the debug registry would always return entries for the management ports, it meant that the debug management ports would always be used instead of the remote cluster ones, meaning that liveness and readiness probes for remote clusters would NEVER work.
```

Comment:
You've got a * which is breaking the link `[server bootup order]*(https://...)` should be `[server bootup order](https://...)`

Reply:
I would elaborate a tiny bit to explain that their are various providers that the aggregate controller uses to get ManagementPorts, and that it iterates over them sequently, taking the first one that returns a non-nil result.
And then further highlight that the MultiCluster one is appended AFTER the debug one, which means that debug will never be called.
You kind of already said that, but just being explicit about how those providers are iterated over i think makes it clearer.

Reply:
Whoa this is really weird. I don't see that in my text editor (the *). Good catch.